external-crates/move/move-compiler/src/editions/mod.rs

```diff
@@ -117,6 +123,13 @@ impl Edition {
         SUPPORTED_FEATURES.get(self).unwrap().contains(&feature)
     }
 
+    pub fn syntax(&self) -> SyntaxEdition {
```

Comment:
why this mapping? I know dumb question with a lot of pragmatic reasons but can you share?

Reply:
Instead of casing on the edition directly you mean? (If this is not what you mean... stop reading :))

Two reasons:
- We want 1 syntax edition for multiple editions, e.g. 2024.alpha 2024.beta will share an edition
- We don't fully enumerate editions, and they are basically just a pair `(Symbol, Option<Symbol>)` so casing on them directly is brittle (and not something we do)